### PR TITLE
Add MultiZarrToZarr.append()

### PR DESCRIPTION
**Pull request by martindurant**
cc @maxrjones

### CODE REVIEW

**Comment by martindurant** [comment]
@rsignell-usgs , have you had a chance to do any experimenting with this?

**Comment by rsignell** [comment]
@martindurant I've (finally) tested and both appending to JSON and Parquet are working fine!

https://nbviewer.org/gist/rsignell/ad1f3c66530a81e77585e05d9683984e

**Comment by martindurant** [comment]
parquet part requires https://github.com/fsspec/filesystem_spec/pull/1510 (should be merged, then released soon)

File: kerchunk/combine.py [hunk 1]
```diff
@@ -348,10 +440,11 @@ def store_coords(self):
     def second_pass(self):
         """map every input chunk to the output"""
         # TODO: this stage cannot be rerun without clearing and rerunning store_coords too,
-        #  because some code runs dependant on the current state f self.out
+        #  because some code runs dependant on the current state of self.out
```
Comment (rsignell) [inline kerchunk/combine.py:443]: dependent

**Review by rsignell — APPROVED** [review]

File: kerchunk/combine.py (hunk 1 above)
Comment (rsignell) [reply kerchunk/combine.py:443]: @maxrjones do you have a few minutes to review?

Comment (martindurant) [reply kerchunk/combine.py:443]: @maxrjones - particularly more tests would be good, although I'm happy to merge this early and add tests as we go

Comment (maxrjones) [reply kerchunk/combine.py:443]: I'll have some time tomorrow afternoon to try it out

Comment (martindurant) [inline kerchunk/combine.py:443]:
```suggestion
        #  because some code runs dependent on the current state of self.out
```

Comment (maxrjones) [reply kerchunk/combine.py]: Sorry, I got caught up with other work but this is still on my list for this week

**Comment by maxrjones** [comment]
@martindurant I added a Pythia chapter on the new feature in https://github.com/ProjectPythia/kerchunk-cookbook/pull/52. It works great, but I got stuck for a while on the need for the `coo_map={"time": "cf:time"}`, since the original MultiZarrtoZarr worked fine without specifying that. Am I missing something regarding how the coordinate alignment works usually?

**Comment by martindurant** [comment]
> need for the coo_map={"time": "cf:time"}

I can't tell you how annoying these time units are in general for someone not from the field. I used to care about leap seconds  https://en.wikipedia.org/wiki/Terrestrial_Time , and barycentring, but it still seemed simpler.

> the original MultiZarrtoZarr worked fine without specifying that

It ... depends. Since the normal mode is to copy the values and attributes as-is, this will be no different to "cf:" and output dtype unspecified, if the inputs happened to have the same time epoch.

**Comment by rsignell** [comment]
So just to make sure I've got it right:   If all the files have the same time units (e.g. "seconds since 1970-01-01 00:00") then we don't need it, but if the files in the collection we are combining have varying time units, like ["hours since 2012-01-01 00:00", "hours since 2013-01-01 00:00"] then we need the map.     Correct?

**Comment by martindurant** [comment]
Correct, but having cf: in the first case shouldn't hurt either (except that you need cftime in the env, which is a given anyway for this case)

**Comment by maxrjones** [comment]
hmm I think in https://github.com/ProjectPythia/kerchunk-cookbook/pull/52 all the files had the same time units, and excluding `cf:` worked in the original concatenation but failed when using append. I created a small test for demonstration in https://github.com/fsspec/kerchunk/pull/414, @martindurant should that be expected to work?

**Pull request merged**